test/markup/handlebars/escaped-mustaches.txt

```diff
@@ -0,0 +1,7 @@
+\{{no-expression}} {{expression}}
+
+\{{{no-expression}}} {{expression}}
+
+\{{#no}} block \{{/no}} {{expression}}
+
+\{{\{{no}}}} block \{{\{{/no}}}} {{expression}}
```

Comment:
What if I want to skip a comment like snippet?  Won't those rules still match?

Reply:
No. You are right, that one is still missing. I'll add that.

Reply:
A `\` in front of the mustache is escaped by another `\` in front of it.
This means that the expressions in 

```handlebars
\\{{expression}}

\\\{{expression}}

\\\\{{expression}}
```
are all evaluated. I have added a commit that skips the escaping if another backslash appears in front of the backslash.

Reply:
I mean don’t the escape rules need to come before the comment rules to allow escaping a comment?

Reply:
I have added a test for escaped comments and it passed without modification of the "handlebars.js" file.

I think, since the match of `\{{` has a lower index than the matching `{{`, it will always be evaluated first.

Also, after writing my own review, I felt I had to extract the rules into there own variables with descriptive names. Sorry for all the changes that you didn't ask about.

I also added consistent escaping of `{` and `}` in the regular expressions of the language. I don't think that is really necessary and I thought about removing all backslashes at first. But since the regex are assembled into larger once, we cannot be sure. There ARE contexts where a `{` needs to be escaped.

Reply:
> I think, since the match of \{{ has a lower index than the matching {{, it will always be evaluated first.

Yeah, I thought I saw the comment regex coming FIRST, which is why i was curious.

Reply:
> Also, after writing my own review, I felt I had to extract the rules into there own variables with descriptive names. 

Do the names have meaning if you know Handlebars really well?  I'm actually not sure it's a win (while normally I'd call it a win)... but to me I SEE the patterns and that  means something to me, but I see the labels you've assigned them and that means nothing to me.

Reply:
If it were me I'd add comment examples beside them.

```
  STANDARD_BLOCK, // {{ ... }}
```